packages/vats/src/localchain.js

```diff
@@ -67,20 +124,23 @@ const prepareLocalChainAccount = zone =>
        * @param {Payment<'nat'>} payment
        * @param {Pattern} [optAmountShape] throws if the Amount of the Payment
        *   does not match the provided Pattern
-       * @returns {Promise<Amount>}
+       * @returns {PromiseVow<Amount<'nat'>>}
        */
       async deposit(payment, optAmountShape) {
-        const { bank } = this.state;
+        const { bank, depositPaymentHandler } = this.state;
 
-        const allegedBrand = await E(payment).getAllegedBrand();
-        const purse = E(bank).getPurse(allegedBrand);
-        return E(purse).deposit(payment, optAmountShape);
+        const getBrandWatcher = watch(E(payment).getAllegedBrand());
+        return watch(getBrandWatcher, depositPaymentHandler, {
```

Comment:
Still learning conventions and thought this was more readable. Aside - it would be awesome to have a style guide / best practices doc for these (naming is tough as well). Happy to apply these changes